### PR TITLE
Blacklist updates

### PR DESCRIPTION
**Pull request by arthurvr**
See https://github.com/gruntjs/gruntjs.com/pull/154#issuecomment-75210828.

@sindresorhus Shall I ping the authors of these into this issue? What do you think?

### CODE REVIEW

File: grunt-plugins.js [hunk 1]
```diff
@@ -22,10 +22,55 @@ var bannedPlugins = [
...
+  'grunt-contrib-levin-usemin', // Reason: unofficial contrib plugin
+  'grunt-contrib-quickstart', // Reason: unofficial contrib plugin
+  'grunt-contrib-i18next', // Reason: unofficial contrib plugin
+  'grunt-contrib-ftpush',  // Reason: unofficial contrib plugin
+  'grunt-mindirect', // Reason: duplicate of contrib-uglify
+  'private-grunt-contrib-uglify', // Reason: duplicate of contrib-uglify
+  'grunt-handlebars-static', // Reason: duplicate of compile-handlebars
+  'grunt-contrib-jshint-reporter-tweaks', // Reason: unofficial contrib plugin
+  'grunt-ftp-deploy', // Reason: duplicate of grunt-ftp
+  'grunt-ftp-deploy', // Reason: duplicate of grunt-ftp
```
Comment (vladikoff) [inline grunt-plugins.js]: listed twice?

Comment (arthurvr) [inline grunt-plugins.js]: fixed

**Comment by sindresorhus** [comment]
Yeah, do that.

**Comment by sindresorhus** [comment]
@vladikoff @tkellen Can either of you deploy this to Heroku?

**Comment by vladikoff** [comment]
Done!

**Comment by arthurvr** [comment]
@amfe @nariyu @trkw @soundasleep @twoer @trenker @vanetix @ninjaYang @zhh77 @dsheiko @donghanji @fkscorpion @dbankier @valeriansaliou @levincao1 @mauvm @sabarasaba @inossidabile @buunguyen @icagstrout @meastes @zonak @andy-shea @dfsq @thanpolas @ngbp @jzaefferer @fweinb @jharding @geta6 @avalade @cbonnissent @geddski @mintbridge @poppinlp @pierreinglebert @makuta Just an FYI. If I blacklisted something without a good reason, sorry! Feel free to complain. If you're wondering why your package was blacklisted, you can take a look at the inline comments. Please also read http://gruntjs.com/creating-plugins#naming-your-task.

**Comment by poppinlp** [comment]
> 'grunt-htmlmin', // Reason: duplicate of contrib-htmlmin

@arthurvr  I made this plugin because i don't wanna do some minify stuff to files which not changed.

**Comment by arthurvr** [comment]
@poppinlp Sorry, I will take a look this evening.

**Comment by mauvm** [comment]
Oops! My logic failed me here. :grin: I figured `grunt-quickstart` would be the official one, and since I came across all kind of `grunt-contrib-` packages, I figured I'd use `contrib` too. I'll go and rename it to `grunt-quickstart`!

**Comment by sabarasaba** [comment]
Sorry, fixed mine already and published it to npm again.

**Comment by zonak** [comment]
Can someone explain why am I included in this???
I do maintain one plugin `grunt-ftp-deploy` and if it is blacklisted somewhere I would like to know why???

**Comment by arthurvr** [comment]
> and if it is blacklisted somewhere I would like to know why???

The answer is right in the diff :)

> Oops! My logic failed me here. :grin: I figured grunt-quickstart would be the official one

Sorry for the confusion.

> Sorry, fixed mine already and published it to npm again.

Yay! Thank you!

**Comment by zonak** [comment]
So let me summarize:

- someone forks the work I did
- never submitted a PR if they thought that this is a good idea
- has 63 start vs the 400 my plugin has
- hasn't had an update since 12/20/2013 - --> **2013** <--
- no one has responded to issues in years

and mine is blacklisted and the other one is proclaimed an official one. Great Job 👏👏👏👏

Awesome way to alienate the community.

**Comment by zonak** [comment]
The last comment was referred to the `contrib` one - didn't realize there is another one.

In either case  the practice is still very bad. Because someone has some authority on a project which is supposed to be a community one - you kick people out like this. You are not Apple to behave like this (and they shouldn't either). Someone did work for the community and you blacklist them in a way there is no remedy???

A W E S O M E.

I'll just stop maintaining the task and pint to this thread ...

**Comment by arthurvr** [comment]
> @arthurvr I made this plugin because i don't wanna do some minify stuff again to files which not changed.

Maybe you should clarify that a little more in your readme (or contribute to grunt-contrib-htmlmin).

**Comment by sindresorhus** [comment]
@poppinlp That's what [`grunt-newer`](https://github.com/tschaub/grunt-newer) is for. Every task shouldn't duplicate that functionality.

**Comment by sindresorhus** [comment]
@zonak People are still free to search npmjs.com for anything. The discovery for grunt plugins is currently really bad and this is a step to improve the situation. There's currently 4338 grunt plugins and a lot of them are useless duplicates. This makes it harder for users to find what they need. Blacklisting has worked very well for `gulp`. We mentioned all of the authors so they're free to complain.

**Comment by vladikoff** [comment]
@zonak https://github.com/gruntjs/gruntjs.com/commit/fb965342fab7160601f548720e3a4d2effdccd32
I reverted the `grunt-ftp-deploy` due to its popularity with 400 GitHub stars. I don't mind listing both plugins on the page. Sorry about that. Just in case anyone else has the same issue, please just let us know and we will fix it up. The blacklisting is mostly for really really bad plugins, not for banning forks that contain different features.

**Comment by zonak** [comment]
I was blamed for my tone and my poor communication - apologies if I did wrong but I am still not sure what was my mistake? Any sane person valuing their work would object if something like this happened to them.
But I digress on this - if my communication was bad, lets see how this action was communicated from your end ... It was not. I just wake up one morning with the notice that my project was blacklisted. Why? Because someone with authority decided so?
You are saying that anyone can search for anything on npmjs.org in the context how this is not a big deal - then why all this??? You are saying that this addresses an issue but then you are saying it is not significant.
I thought it is common that overlaps should be resolved with a discussion or cooperation - but no, you decided to blacklist it and point to a project that doesn't cover all of the features.
There are lots of _ftp_ and _deploy_ related projects that are not blacklisted - so the rules don't seem to be that clear.
Further more I am still confused on what are my options? I don't see anywhere someone saying anything about how can this be changed. It is just a done thing.

As I was writing this i saw the post by @vladikoff - thank you for what you did there. But I still think that this is not the right way to do things. There should have been an issue started like this where people would have been given some time and oportunity to voice their opinion before the action was taken.

**Comment by zonak** [comment]
@vladikoff with the only purpose of trying to better understanding this, I wanted to ask about:

> not for banning forks that contain different features.

not sure what you meant by this but my plugin is not a fork and even if it was it should still get at least the courtesy of discussion.

**Comment by cowboy** [comment]
Instead of blacklisting these plugins, why don't we just note to the users that they aren't official contrib plugins, and let them decide?

**Comment by cowboy** [comment]
I mean, it's not like we can blacklist them from npm too.

**Comment by vladikoff** [comment]
@zonak Ah sorry, i read this wrong:

> someone forks the work I did

Really sorry again for blacklisting the plugin, we do appreciate plugin contributions and it is a great time saver for the community. With a simple fix to the list the issue is fixed and the site was redployed. I'm gonna review the new list that was merged here and see if there are other plugins that were not properly marked. If anyone else has issues with their plugins please create an issue in the issue tracker.

**Pull request merged**